### ISSUE TITLE
rf64_downgrade_test fails with VS2017

### ISSUE THREAD

**Issue by evpobr** [erikd/libsndfile#244]
Run tests today - rf64_downgrade_test fails:

```
test 72
    Start 72: misc_test_rf64

72: Test command: C:\Users\evpob\Documents\GitHub\libsndfile\CMakeBuild\Release\misc_test.exe "rf64"
72: Test timeout computed to be: 1500
72:     zero_data_test                 : zerolen.rf64 ............ ok
72:     rf64_downgrade_test            : downgrade.wav ...........
72:
72: Line 458: sf_open (SFM_READ) failed : Error in WAV file. No 'data' chunk marker.
72:
72: File : downgrade.wav
72: Length : 2160
72: RIFF : 2152
72: WAVE
72: JUNK : 24
72: *** Unknown chunk marker (100016) at position 44. Exiting parser.
72:
1/1 Test #72: misc_test_rf64 ...................***Failed    0.03 sec
```

Call stack:

```
>	misc_test.exe!psf_binheader_writef(sf_private_tag * psf, const char * format, ...) Line 752	C
 	misc_test.exe!rf64_write_header(sf_private_tag * psf, int calc_length) Line 669	C
 	misc_test.exe!sf_write_short(SNDFILE_tag * sndfile, const short * ptr, __int64 len) Line 2171	C
 	misc_test.exe!test_write_short_or_die(SNDFILE_tag * file, int pass, const short * test, __int64 items, int line_num) Line 794	C
 	misc_test.exe!rf64_downgrade_test(const char * filename) Line 450	C
 	misc_test.exe!main(int argc, char * * argv) Line 150	C

```

**Comment by erikd** [comment]
That's with current git HEAD?

**Comment by evpobr** [comment]
Yes, it is master. Win10 64-bit.

**Comment by erikd** [comment]
This will likely be fixed with #241 which I'm currently working on.

**Comment by evpobr** [comment]
@erikd, test is [passed](https://ci.appveyor.com/project/Erik13183/libsndfile/build/job/8a1uorqcu3jy1gy6) in 32-bit mode.

**Comment by erikd** [comment]
I'm not surprised. Its related to #241 and the diffiering sizes of types across platforms and 32 vs 64 bit versions of those platforms.

**Comment by evpobr** [comment]
Fixed via #299
